core/arch/arm/kernel/link.mk

```diff
@@ -219,6 +219,9 @@ $(link-out-dir)/tee.bin: $(link-out-dir)/tee-pager.bin \
 		--init_mem_usage `cat $(link-out-dir)/tee-init_mem_usage.txt` \
 		--tee_pager_bin $(link-out-dir)/tee-pager.bin \
 		--tee_pageable_bin $(link-out-dir)/tee-pageable.bin \
+		--tee_header_v2_out $(link-out-dir)/tee-header-v2.bin\
+		--tee_pager_v2_out $(link-out-dir)/tee-pager-v2.bin\
+		--tee_pageable_v2_out $(link-out-dir)/tee-pageable-v2.bin\
```

Comment:
- Please add a space before the backslashes
- `*-v2.bin` are output files, so:
  - they should appear as targets somehow
  - they should be added to $(cleanfiles)
  - they should be added as prerequisites of `all:`
- I think it's best to invoke `gen_hashed_bin.py` once per output file. All sorts of problems may arise otherwise, i.e., when a command generates several output files (for instance, `make -jN`).

So here is how I would write the rules:

```make
gen_hash_bin_deps := $(link-out-dir)/tee-pager.bin \
                         $(link-out-dir)/tee-pageable.bin \
                         $(link-out-dir)/tee-init_size.txt \
                         $(link-out-dir)/tee-init_load_addr.txt \
                         $(link-out-dir)/tee-init_mem_usage.txt \
                        ./scripts/gen_hashed_bin.py
define gen_hash_bin_cmd
	@$(cmd-echo-silent) '  GEN     $@'
	$(q)load_addr=`cat $(link-out-dir)/tee-init_load_addr.txt` && \
	./scripts/gen_hashed_bin.py \
		--arch $(if $(filter y,$(CFG_ARM64_core)),arm64,arm32) \
		--init_size `cat $(link-out-dir)/tee-init_size.txt` \
		--init_load_addr_hi $$(($$load_addr >> 32 & 0xffffffff)) \
		--init_load_addr_lo $$(($$load_addr & 0xffffffff)) \
		--init_mem_usage `cat $(link-out-dir)/tee-init_mem_usage.txt` \
		--tee_pager_bin $(link-out-dir)/tee-pager.bin \
		--tee_pageable_bin $(link-out-dir)/tee-pageable.bin
endef

all: $(link-out-dir)/tee.bin
cleanfiles += $(link-out-dir)/tee.bin
$(link-out-dir)/tee.bin: $(gen_hashed_bin_deps)
	$(gen_hash_bin_cmd) --out $@

all: $(link-out-dir)/tee-header-v2.bin
cleanfiles += $(link-out-dir)/tee-header-v2.bin
$(link-out-dir)/tee-header-v2.bin: $(gen_hashed_bin_deps)
	$(gen_hash_bin_cmd) --tee_header_v2_out $@

# And same for the two other v2 binaries
```